TAGradingServer/account/admin-gradeable.php

```diff
@@ -881,6 +881,26 @@ function deleteForm() {
             return false;
         }
     }
+
+    function createCrossBrowserJSDate(val){
+        // Create a Date object that is cross-platform supported.
+        // Safari's Date object constructor is more restrictive that Chrome and 
+        // Firefox and will treat some dates as invalid.  Implementation details
+        // vary by browser and JavaScript engine.
+        // To solve this, we use Moment.js to standardize the parsing of the 
+        // datetime string and convert it into a RFC2822 / IETF date format.
+        //
+        // For example, ""2013-05-12 20:00:00"" is converted with Moment into 
+        // "Sun May 12 2013 00:00:00 GMT-0500 (EST)" and correctly parsed by
+        // Safari, Chrome, and Firefox.
+        //
+        // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse
+        // Ref: http://stackoverflow.com/questions/16616950/date-function-returning-invalid-date-in-safari-and-firefox
+        var timeParseString = "YYYY-MM-DD HH:mm:ss.S" // Expected string format given by the server
+        var momentDate = moment(val, timeParseString) // Parse raw datetime string
+        return new Date(momentDate.toString()) // Convert moment into RFC2822 and construct browser-specific jQuery Date object
```

Comment:
I agree, yet Moment allows for some nice Date comparisons and operations that might make working with  liner-side date restrictions more convenient.  I found this string replace fix as well but opted for a more powerful solution that gives us much more flexibility in terms of manipulating dates. 

For example, we could easily script the other dates to initially offset based on a fixed number of days from the due date.  We could also do some very thorough client side sanity checks before submission with an opportunity for the user to correct their mistake.   The dates could be local specific and provide easy support for different time zones when Submitty is extended at other universities.  We could use a framework like Moment that has robust date parsing and logic as a foundation for this additional client-side code. 

I don't mind using the simple string replace, it's very much a simpler solution.  However, there is merit in doing more advanced processing of the dates client-side before the gradeable form is submitted to the server and validated.